shared/actions/chat.js

```diff
@@ -527,6 +548,12 @@ function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
           // have, just set it to failed.  If we haven't, record this as a
           // pending failure, and pick up the pending failure at the bottom of
           // _postMessage() instead.
+          //
+          // Do we have this conversation loaded?  If not, don't do anything -
+          // we'll pick up the failure when we load that thread.
+          const isConversationLoaded = yield select(_conversationStateSelector, conversationIDKey)
+          if (!isConversationLoaded) return
+
```

Comment:
This change is unrelated to the purpose of the PR.  I discovered that incoming FailedMessage notifications can reference messages that we don't have loaded in our conversationStates, because they're from a previous app run.  So we check whether the conversation referenced in a failed message notification is loaded.